azkaban-web-server/src/main/java/azkaban/webapp/AzkabanWebServer.java

```diff
@@ -510,6 +511,13 @@ private void addThreadPoolGauges(final QueuedThreadPool threadPool) {
   private void startWebMetrics() throws Exception {
     this.metricsManager
         .addGauge("WEB-NumQueuedFlows", this.executorManagerAdapter::getQueuedFlowSize);
+
+    // Metric for flows that have been submitted, but haven't started for more than 20 minutes.
+    if (this.props.getBoolean(ConfigurationKeys.AZKABAN_POLL_MODEL, false)) {
```

Comment:
Would your change work as-is for the older push-based dispatch model? If yes, then should we extend this to include that ? (there will be some testing overhead I suppose) as I believe that model is still used by a couple of instances (and as also more in-line with future containerization related work).
Doesn't necessarily have to be part of this PR, but worth checking.

Reply:
Sure. Makes sense, I will look into this. 
There is a 2nd part to this PR. So can check this out too.

The reason, the metric for new dispatch logic is more important is because if recent fixes that are supposed to improve the situation of flows getting stuck in PREPARING state - it will be good to have a metric that measures that aspect well.